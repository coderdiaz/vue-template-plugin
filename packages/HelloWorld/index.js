import HelloWorld from './src/main.vue';

HelloWorld.install = (Vue) => {
  Vue.component(HelloWorld.name, HelloWorld);
};

export default HelloWorld;
