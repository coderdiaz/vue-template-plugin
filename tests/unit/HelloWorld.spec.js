import { shallowMount } from '@vue/test-utils';
import HelloWorld from '../../packages/HelloWorld/src/main.vue';

// Helper function to create a component
const createComponent = propsData => shallowMount(HelloWorld, { propsData });

// Mounting an instance of HelloWorld component
const vm = shallowMount(HelloWorld);

describe('HelloWorld.vue', () => {
  let cmp;
  it('should match the snapshot', () => {
    expect(vm.$el).toMatchSnapshot();
  });

  it('HelloWorld is a component', () => {
    expect(vm.isVueInstance()).toBeTruthy();
  });

  describe('Properties', () => {
    it('when the component is created without `message` prop, the component shows the message by default', () => {
      cmp = createComponent();
      expect(cmp.vm.message).toBe('Hello World');
    });

    it('when the components is created with a custom message, the component shows the custom message', () => {
      cmp = createComponent({
        message: 'Hello my name is Javier',
      });
      expect(cmp.vm.message).toBe('Hello my name is Javier');
    });
  });
});
