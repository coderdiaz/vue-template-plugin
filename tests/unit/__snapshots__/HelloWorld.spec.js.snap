// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue should match the snapshot 1`] = `undefined`;
